python/ray/tune/trainable/function_trainable.py

```diff
@@ -50,6 +51,7 @@ def setup(self, config):
                 logdir=self._storage.trial_driver_staging_path,
                 driver_ip=None,
                 experiment_name=self._storage.experiment_dir_name,
+                run_id=uuid.uuid4().hex,
```

Comment:
Do we want/need to handle this throughout Tune as well? Should we limit this to Train?

Reply:
I'm trying to limit this to Train, but found this is the cleanest way to do it (be consistent with existing ones: trial_id, trial_name, exp_name, ...). It introduces minimal code change by using `_TrainSession` to propagate and retrieved it in both trainer and workers.